src/object.js

```diff
@@ -20,233 +23,700 @@ module.exports = (common) => {
   })
 
   describe('.object', () => {
-    const testObject = Buffer(JSON.stringify({Data: 'testdata', Links: []}))
-    const testObjectHash = 'QmPTkMuuL6PD8L2SwTwbcs1NPg14U8mRzerB1ZrrBrkSDD'
-    const testPatchObject = Buffer(JSON.stringify({Data: 'new test data'}))
-    const testPatchObjectHash = 'QmWJDtdQWQSajQPx1UVAGWKaSGrHVWdjnrNhbooHP7LuF2'
+    it('object.new', (done) => {
+      ipfs.object.new((err, node) => {
+        expect(err).to.not.exist
+        expect(node.toJSON().Hash).to.equal('QmdfTbBqBPQ7VNxZEYEj14VmRuZBkqFbiwReogJgS1zR1n')
+        done()
+      })
+    })
 
-    it('object.put', (done) => {
-      ipfs.object.put(testObject, 'json', (err, res) => {
+    it('object.put of object', (done) => {
+      const obj = {
+        Data: new Buffer('Some data'),
+        Links: []
+      }
+
+      ipfs.object.put(obj, (err, node) => {
         expect(err).to.not.exist
-        expect(res).to.have.a.property('Hash', testObjectHash)
-        expect(res.Links).to.be.empty
+        const nodeJSON = node.toJSON()
+        expect(obj.Data).to.deep.equal(nodeJSON.Data)
+        expect(obj.Links).to.deep.equal(nodeJSON.Links)
+        expect(nodeJSON.Hash).to.equal('QmPb5f92FxKPYdT3QNBd1GKiL4tZUXUrzF4Hkpdr3Gf1gK')
         done()
       })
     })
 
-    it('object.get', (done) => {
-      ipfs.object.get(testObjectHash, (err, res) => {
+    it('object.put of json encoded buffer', (done) => {
+      const obj = {
+        Data: new Buffer('Some data').toString(),
+        Links: []
+      }
+
+      const buf = new Buffer(JSON.stringify(obj))
+
+      ipfs.object.put(buf, { enc: 'json' }, (err, node) => {
         expect(err).to.not.exist
-        expect(res).to.have.a.property('Data', 'testdata')
-        expect(res.Links).to.be.empty
+        const nodeJSON = node.toJSON()
+        expect(obj.Data).to.deep.equal(nodeJSON.Data)
+        expect(obj.Links).to.deep.equal(nodeJSON.Links)
+        expect(nodeJSON.Hash).to.equal('QmPb5f92FxKPYdT3QNBd1GKiL4tZUXUrzF4Hkpdr3Gf1gK')
         done()
       })
     })
 
-    it('object.data', (done) => {
-      ipfs.object.data(testObjectHash, (err, res) => {
+    // TODO verify that yaml encoded buffers still work in go-ipfs
+    it.skip('object.put of yaml encoded buffer', (done) => {
```

Comment:
@diasdavid yaml was never available, json and protobuf are the supported encodings according to the help section
```
   Data should be in the format specified by the --inputenc flag.
    --inputenc may be one of the following:
    	* "protobuf"
    	* "json" (default)
```